core/metadata-3.2.md

```diff
@@ -121,6 +155,146 @@ using `ERR_KEYNOPERMISSION` with an asterisk (`*`) in the `<Key>` field.
+client remains subscribed to the key. In case such a key is processed in a
+request the key MUST appear at least once in a `RPL_METADATASUBOK` numeric in
+the reply.
+
+Servers MUST respond to requests to subscribe to a key whose name is invalid
+with a `ERR_KEYINVALID` numeric.
+
+Servers MAY additionally respond to requests to subscribe to a key that the
+client has no privilege to access with a `ERR_KEYNOPERMISSION` numeric.
+Even if a server does that, the subscription MUST still be successful.
+This is because in this case the `ERR_KEYNOPERMISSION` numeric only serves as a
+warning, indicating that the client will not receive METADATA events about this
+key unless it acquires the necessary (implementation defined) privileges later.
+
+### METADATA UNSUB
+
+This subcommand is used to unsubscribe from metadata keys.
+
+Syntax:
+
+    METADATA * UNSUB <key1> [<key2> ...]
+
+The reply of the server MUST be zero or more numerics of the following types
+in any order: `RPL_METADATAUNSUBOK`, `ERR_KEYINVALID`.
+Then the server MUST end the reply with one `RPL_METADATAEND` numeric.
+
+If a client successfully unsubscribes from a key it is subscribed to then the
+key MUST appear in a `RPL_METADATAUNSUBOK` reply numeric.
+
+If a client tries to unsubscribe from a key that it is not subscribed to then
+the client remains not subscribed to the key and the key MUST appear at least
+once in a `RPL_METADATAUNSUBOK` numeric in the reply.
+
+Servers MUST respond to requests to subscribe to a key whose name is invalid
+with a `ERR_KEYINVALID` numeric.
+
+### METADATA SUBS
+
+This subcommand can be used to get a list of keys which the client is
+subscribed to.
+
+Syntax:
+
+    METADATA * SUBS
+
+The server MUST reply with zero or more `RPL_METADATASUBS` numerics and then
+one `RPL_METADATAEND` numeric.
+
+The replied `RPL_METADATASUBS` numerics, collectively, MUST contain all keys
+the client is subscribed to exactly once and MUST NOT contain keys the client
+is not subscribed to.
+
+### METADATA SYNC
+
+This subcommand requests the full synchronization of metadata associated with
+the given target.
+
+Syntax:
+
+    METADATA <Target> SYNC
+
+The result of this subcommand is either zero or more METADATA events on
+success, or a `ERR_METADATASYNCLATER` numeric if the synchronization cannot be
+performed at this time.
+
+For details, please see the Postponed synchronization subsection of the
+Metadata notifications section.
+
+### RPL_METADATASUBS
+
+The order of the keys is undefined.
+
+### ERR_METADATATOOMANYSUBS
+
+The `<key>` parameter of this numeric is the first key that the client was not
+subscribed to.
+
+## Notification Mechanics (TODO merge with Metadata Notifications)
+
+A client can either be subscribed to a key, or not subscribed to it.
+
+The server MUST allow a client to subscribe to any valid keys, even to
+privileged keys when the client has no privilege to access that key at
+the time of subscription.
```

Comment:
Why?

Reply:
Presumably because the privilege may change and then they can start getting notifications.

Reply:
Makes sense to me. If the key name's valid, just throw it in the sync list no matter what. For example, if they setup all their metadata stuff on connecting and then after connect they oper up.